shell/browser/ui/file_dialog_win.cc

```diff
@@ -65,26 +65,15 @@ void ConvertFilters(const Filters& filters,
   }
 }
 
-static HRESULT GetFileNameFromShellItem(IShellItem* pShellItem,
-                                        SIGDN type,
-                                        LPWSTR lpstr,
-                                        size_t cchLength) {
-  assert(pShellItem != nullptr);
-
-  LPWSTR lpstrName = nullptr;
-  HRESULT hRet = pShellItem->GetDisplayName(type, &lpstrName);
-
-  if (SUCCEEDED(hRet)) {
-    if (wcslen(lpstrName) < cchLength) {
-      wcscpy_s(lpstr, cchLength, lpstrName);
-    } else {
-      NOTREACHED();
-    }
+auto GetFileNameFromShellItem(IShellItem* pShellItem, SIGDN type) {
+  std::optional<base::FilePath> path;
 
-    ::CoTaskMemFree(lpstrName);
+  if (wchar_t* name = {}; SUCCEEDED(pShellItem->GetDisplayName(type, &name))) {
+    path.emplace(name);
```

Comment:
@ckerr we are freeing the memory pointed by name in the next line, shouldn't this copy stuff into `path` ?

Reply:
@deepak1556 I might be missing something but I _think_ this code is correct as-is:

[std::optional&lt;base::FilePath&gt;::emplace()](https://en.cppreference.com/w/cpp/utility/optional/emplace) is going to call the `base::FilePath` [constructor](https://chromium.googlesource.com/chromium/src/+/main/base/files/file_path.h#204), which takes a string view. So `path` here has its own copy of `name` and `name` is safe to be freed on line 73